2024/XenSummit/Challenges_and_Status_of_Enabling_TrenchBoot_in_Xen_Hypervisor.md

```diff
@@ -0,0 +1,310 @@
+  aid against [Evil Maid attacks](https://theinvisiblethings.blogspot.com/2009/10/evil-maid-goes-after-truecrypt.html)
+- Leverages DRTM and TPM to seal secrets, which are used by the owner to
+  confirm whether the device has been tampered with or not
+- Initially only Intel TXT and TPM1.2 with tboot was supported using legacy
+  BIOS boot mode, now TrenchBoot replaces tboot, extending the support to
+  TPM2.0 and AMD SKINIT
+
+---
+
+# Demo: Qubes OS Anti Evil Maid
+
+.left-column50[
+
+<a href="https://asciinema.org/a/661622?cols=80&rows=24"
+target="_blank"> <img src="https://asciinema.org/a/661622.svg" width="400px"/></a>
+
+]
+
+.right-column45[
+
+- **Hardware**: Protectli VP4670
+- **TPM**: Infineon SLB9665 LPC TPM 2.0
+- **Firmware**: Dasharo (coreboot+SeaBIOS) v0.9.0
+  * coreboot 24.02.01
+  * SeaBIOS rel-1.16.3
+- **Issue**: [Xen is booting very slowly after Dynamic
+  Launch](https://github.com/TrenchBoot/xen/issues/16)
+
+]
+
+.footnote[
+Watch on [asciinema](https://asciinema.org/a/661622?cols=80&rows=24)
+]
+
+---
+
+# Current status of the work
+
+.center[.image-99[![](/img/aem_progress_2024.png)]]
+
+.center[.image-40[![](/img/logo/nlnet.svg)]]
+
+- https://github.com/TrenchBoot/trenchboot-issues/milestones
+
+???
+
+The project, sponsored by NLnet, is split into phases, where each phase builds
+on the previous one and enables more usage scenarios.
+
+Since the last year we progressed pretty significantly and finalize subsequent
+3 phases.
+
+---
+
+# Phase 2 - TPM 2.0 support
+
+.center[.image-80[![](/img/tb_aem_phase2.png)]]
+
+???
+
+- Added TPM 2.0 support for dom0 kernel and initrd measurements
+- Added eventlog extraction and parsing
+- Implemented support for booting with Intel TXT and TPM 2.0 in legacy BIOS
+  boot mode
+
+---
+
+# Phase 3 - TrenchBoot boot protocol
+
+.center[.image-99[![](/img/tb_aem_phase3.png)]]
+
+???
+
+- Rebased changes onto the most recent TrenchBoot support patches to Linux
+  kernel introducing a unified TrenchBoot boot protocol
+- Final shape of patches is not yet determined, the work on finalizing the
+  boot protocol including UEFI boot mode is pending on the [Linux kernel
+  mailing list, currently patchset
+  v8](https://lore.kernel.org/lkml/8d543a15-af62-4403-b2e0-3b395edfe9e4@amd.com/T/)
+
+---
+
+# Phase 4 - AMD support
+
+.center[.image-90[![](/img/tb_aem_phase4.png)]]
+
+???
+
+- Implemented support for booting with AMD SKINIT and TPM 1.2/2.0 in legacy
+  BIOS boot mode
+- Implemented the newest TrenchBoot boot protocol in the Secure Kernel Loader
+- Simplified installation of the required packages by providing a RPM
+  repository for QubesOS
+
+---
+
+# Try TenchBoot
+
+It is relatively easy to get a hardware which supports DRTM:
+
+- Intel-based tested and known to work:
+  * Protectli VP4670
+  * HP EliteDesk 800 G2 Desktop Mini
+
+.center[.image-30[![](/img/VP4600.png)].image-30[![](/img/hp_elitedesk_800_g2.jpg)]]
+
+- AMD-based tested and known to work:
```

Comment:
KGPE technically worked with TrenchBoot, it is Xen/QubesOS combo that doesn't like such old CPU.

Reply:
Technically yes, but yet it really didn't... I know it is generally Xen/Qubes OS combo that is too heavy for KGPE and TrenchBoot has nothing to do with it. However, I don't want to put that here for this particular reason, because it doesn't work well in the end.